### PR TITLE
Add metric name in limiter per-metric exceeded errors

### LINKED ISSUE #6416: Include metric name in max series per metric limit exceeded error
**Is your feature request related to a problem? Please describe.**
When receiving an errMaxSeriesPerMetricLimitExceeded error, the metric name is not displayed.

**Describe the solution you'd like**
Include the metric name in the error message.

**Additional context**
When a user hits the max-series per metric name limit, they should know which metric caused the limit to be reached so that they can reduce their active series for that limit or increase the limit.

### PR DESCRIPTION
**Pull request by eeldaly**
**What this PR does**:
Adds metric name in limiter per-metric exceeded errors

**Which issue(s) this PR fixes**:
Fixes #6416 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,11 +46,12 @@
 * [ENHANCEMENT] Add new option `-server.grpc_server-num-stream-workers` to configure the number of worker goroutines that should be used to process incoming streams. #6386
 * [ENHANCEMENT] Distributor: Return HTTP 5XX instead of HTTP 4XX when instance limits are hit. #6358
 * [ENHANCEMENT] Ingester: Make sure unregistered ingester joining the ring after WAL replay. #6277
+* [ENHANCEMENT] Distributor: Add a new `-distributor.num-push-workers` flag to use a goroutine worker pool when sending data from distributor to ingesters. #6406
+* [ENHANCEMENT] Added metric name in limiter per-metric exceeded errors
```
Comment (yeya24) [inline CHANGELOG.md:50]: Can we include the PR number here?

File: pkg/ingester/limiter.go [hunk 2]
```diff
@@ -67,22 +68,22 @@ func NewLimiter(
...
 // AssertMaxSeriesPerMetric limit has not been reached compared to the current
 // number of series in input and returns an error if so.
-func (l *Limiter) AssertMaxSeriesPerMetric(userID string, series int) error {
+func (l *Limiter) AssertMaxSeriesPerMetric(userID string, series int, metric string) error {
 	if actualLimit := l.maxSeriesPerMetric(userID); series < actualLimit {
 		return nil
 	}
 
-	return errMaxSeriesPerMetricLimitExceeded
+	return errors.Wrap(errMaxSeriesPerMetricLimitExceeded, "{metric name: " + metric + "}")
```
Comment (yeya24) [inline pkg/ingester/limiter.go:76]: Can we change `errMaxSeriesPerMetricLimitExceeded` to take metric name as a parameter and we use `errors.Wrapf(errMaxSeriesPerMetricLimitExceeded, metric)`?

Comment (alanprot) [reply pkg/ingester/limiter.go]: We already have the labels on the `handleAppendFailure`, no? we can just send those labels down to format the error?

Comment (alanprot) [reply pkg/ingester/limiter.go]: Ex: https://github.com/cortexproject/cortex/blob/6be037866e7b71706628b93fe3953129230e548e/pkg/ingester/ingester.go#L1161

File: pkg/ingester/limiter.go [hunk 3]
```diff
@@ -176,16 +176,16 @@ func (l *Limiter) formatMaxMetadataPerUserError(userID string) error {
...
 	globalLimit := l.limits.MaxGlobalMetadataPerMetric(userID)
 
-	return fmt.Errorf("per-metric metadata limit of %d exceeded, %s (local limit: %d global limit: %d actual local limit: %d)",
-		minNonZero(localLimit, globalLimit), l.AdminLimitMessage, localLimit, globalLimit, actualLimit)
+	return fmt.Errorf("per-metric metadata limit of %d exceeded for metric %s, %s (local limit: %d global limit: %d actual local limit: %d)",
+		minNonZero(localLimit, globalLimit), lbls.Get(labels.MetricName), l.AdminLimitMessage, localLimit, globalLimit, actualLimit)
 }
 
-func (l *Limiter) formatMaxSeriesPerLabelSetError(err errMaxSeriesPerLabelSetLimitExceeded) error {
+func (l *Limiter) formatMaxSeriesPerLabelSetError(err errMaxSeriesPerLabelSetLimitExceeded, lbls labels.Labels) error {
```
Comment (yeya24) [inline pkg/ingester/limiter.go:188]: This seems not used. Do we need it?

File: pkg/ingester/limiter.go [hunk 4]
```diff
@@ -158,16 +158,16 @@ func (l *Limiter) formatMaxSeriesPerUserError(userID string) error {
...
 	globalLimit := l.limits.MaxGlobalSeriesPerMetric(userID)
 
-	return fmt.Errorf("per-metric series limit of %d exceeded, %s (local limit: %d global limit: %d actual local limit: %d)",
-		minNonZero(localLimit, globalLimit), l.AdminLimitMessage, localLimit, globalLimit, actualLimit)
+	return fmt.Errorf("per-metric series limit of %d exceeded for metric %s, %s (local limit: %d global limit: %d actual local limit: %d)",
+		minNonZero(localLimit, globalLimit), lbls.Get(labels.MetricName), l.AdminLimitMessage, localLimit, globalLimit, actualLimit)
 }
 
-func (l *Limiter) formatMaxMetadataPerUserError(userID string) error {
+func (l *Limiter) formatMaxMetadataPerUserError(userID string, lbls labels.Labels) error {
```
Comment (yeya24) [inline pkg/ingester/limiter.go:170]: This seems not used. Do we need it?

File: pkg/ingester/limiter.go [hunk 5]
```diff
@@ -130,26 +130,26 @@ func (l *Limiter) AssertMaxSeriesPerLabelSet(userID string, metric labels.Labels
...
 		errors.As(err, &e)
-		return l.formatMaxSeriesPerLabelSetError(e)
+		return l.formatMaxSeriesPerLabelSetError(e, lbls)
 	default:
 		return err
 	}
 }
 
-func (l *Limiter) formatMaxSeriesPerUserError(userID string) error {
+func (l *Limiter) formatMaxSeriesPerUserError(userID string, lbls labels.Labels) error {
```
Comment (yeya24) [inline pkg/ingester/limiter.go:152]: This seems not used. Do we need it?

File: pkg/ingester/limiter.go [hunk 6]
```diff
@@ -158,16 +158,16 @@ func (l *Limiter) formatMaxSeriesPerUserError(userID string) error {
 		minNonZero(localLimit, globalLimit), l.AdminLimitMessage, localLimit, globalLimit, actualLimit)
 }
 
-func (l *Limiter) formatMaxSeriesPerMetricError(userID string) error {
+func (l *Limiter) formatMaxSeriesPerMetricError(userID string, lbls labels.Labels) error {
 	actualLimit := l.maxSeriesPerMetric(userID)
```
Comment (yeya24) [inline pkg/ingester/limiter.go:162]: We can just pass the metric name string. No need to pass labels

**Review by yeya24 — APPROVED** [review]
LGTM. Let's fix lint

**Review by alanprot — APPROVED** [review]
LGTM

**Pull request merged**